js/libs/builder.js

```diff
@@ -323,6 +323,44 @@ Fliplet().then(function() {
           Fliplet.Studio.emit('widget-mode', 'wide');
         }
       },
+      getDataSourceColumnValues: function(field) {
+        var id = field.dataSourceId;
+        var column = field.column;
+
+        if (!id || !column) {
+          return;
+        }
+
+        Fliplet.Cache.get({
+          key: id + '-' + column,
+          expire: 60
+        }, function getColumnValues() {
+          // If there's no cache, return new values, i.e.
+          Fliplet.DataSources.connect(id).then(function(connection) {
+            connection.getIndex(column).then(function onSuccess(values) {
```

Comment:
Doesn't the request need to be returned for the promise to work?

Also, the `.then()` can be chained instead of nested.

```suggestion
            return connection.getIndex(column);
          }).then(function onSuccess(values) {
```